tests/ttnn/unit_tests/operations/test_conv3d.py

```diff
@@ -0,0 +1,495 @@
+            )
+
+    assert device.num_program_cache_entries() == 2
+
+
+@pytest.mark.parametrize(
+    "input_shape, out_channels, kernel_size, stride, padding, padding_mode",
+    [
+        [(1, 64, 16, 16, 16), 64, (3, 3, 3), (1, 1, 1), (0, 1, 1), "replicate"],
+    ],
+)
+@pytest.mark.parametrize("grid_size", [[1, 1], [1, 8], [8, 8]], ids=["grid_1x1", "grid_1x8", "grid_8x8"])
+def test_conv3d_multicore(
+    device, input_shape, out_channels, kernel_size, stride, padding, padding_mode, grid_size, use_program_cache
+):
+    # Test that program cache does not re-use the same program for different inputs
+    run_conv3d_test(device, input_shape, out_channels, kernel_size, stride, padding, padding_mode, grid_size=grid_size)
+
+
+@pytest.mark.parametrize(
+    "input_shape, out_channels, kernel_size, stride, padding, padding_mode, blocking",
+    [
+        [
+            (1, 768, 4, 60, 106),
+            768,
+            (3, 3, 3),
+            (1, 1, 1),
+            (0, 1, 1),
+            "replicate",
+            (128, 96, 1, 2, 16),
+        ],  # Best blocking found so far
+        [
+            (1, 512, 11, 120, 212),
+            512,
+            (3, 3, 3),
+            (1, 1, 1),
+            (0, 1, 1),
+            "replicate",
+            (128, 128, 1, 8, 4),
+        ],  # Best blocking found so far
+        [
+            (1, 256, 21, 240, 424),
+            256,
+            (3, 3, 3),
+            (1, 1, 1),
+            (0, 1, 1),
+            "replicate",
+            (128, 128, 4, 4, 2),
+        ],  # Best blocking found so far
+        [
+            (1, 128, 21, 480, 848),
+            128,
+            (3, 3, 3),
+            (1, 1, 1),
+            (0, 1, 1),
+            "replicate",
+            (128, 128, 1, 2, 16),
+        ],  # Best blocking found so far
+    ],
+    ids=["variant1", "variant2", "variant3", "variant4"],
+)
+@pytest.mark.parametrize("grid_size", [[8, 8]], ids=["grid_8x8"])
+def test_conv3d_mochi_shapes(
+    device,
+    input_shape,
+    out_channels,
+    kernel_size,
+    stride,
+    padding,
+    padding_mode,
+    blocking,
+    grid_size,
+    use_program_cache,
+):
+    C_in_block, C_out_block, T_out_block, H_out_block, W_out_block = blocking
+    tt_input, conv3d_module, gt_output, kernel_config, output_dims = setup_conv3d_test(
+        input_shape, out_channels, kernel_size, stride, padding, padding_mode, device, debug=False
+    )
+    N, D_out, H_out, W_out = output_dims
+    C = input_shape[1]
+
+    # Prepare weights with specified C_in_block
+    tt_weight, tt_bias = prepare_weights(conv3d_module, C, out_channels, device, C_in_block=C_in_block)
+
+    config = create_conv3d_config(
+        out_channels,
+        kernel_size,
+        stride,
+        padding,
+        padding_mode,
+        T_out_block=T_out_block,
+        H_out_block=H_out_block,
+        W_out_block=W_out_block,
```

Comment:
Tests in post commit need to be reasonable fasts (think seconds) if this takes 7min to run 
move it to nightly pipeline and make a short sanity check tests for post commit